diesel/src/row.rs

```diff
@@ -154,15 +154,15 @@ mod private {
         feature = "i-implement-a-third-party-backend-and-opt-into-breaking-changes",
         cfg(feature = "i-implement-a-third-party-backend-and-opt-into-breaking-changes")
     )]
-    pub trait RowGatWorkaround<'a, DB: Backend> {
+    pub trait RowSealed<DB: Backend> {
         /// Field type returned by a `Row` implementation
         ///
         /// * Crates using existing backend should not concern themself with the
         ///   concrete type of this associated type.
         ///
         /// * Crates implementing custom backends should provide their own type
         ///   meeting the required trait bounds
-        type Field: Field<'a, DB>;
+        type Field<'a>: Field<'a, DB>;
```

Comment:
why doesn't this move on the `Row` trait?

Reply:
I had to keep the associated type on RowSealed in order to express FieldRet without the lifetime of Row

Reply:
explained in the commit message and additional doc

Reply:
Can you check if `pub type FieldRet<'a, R, DB> = <R as Row<'a, DB>>::Field<'a>;` works for all provided cases? If that's the case I would rather move the `Field` GAT to `Row` instead of having another trait for this.